statistics/databuffer.go

```diff
@@ -0,0 +1,554 @@
+		results = append(results, metric)
+	}
+
+	if flag.NinthPercentile {
+		ninthpercentile = d.PercentileNearestRank(9)
+		metric = createMetric(ninthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("9%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.TwentyFifthPercentile {
+		twentyfifthpercentile = d.PercentileNearestRank(25)
+		metric = createMetric(twentyfifthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("25%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.SeventyFifthPercentile {
+		seventyfifthpercentile = d.PercentileNearestRank(75)
+		metric = createMetric(seventyfifthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("75%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyFirstPercentile {
+		ninetyfirstpercentile = d.PercentileNearestRank(91)
+		metric = createMetric(ninetyfirstpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("91%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyEighthPercentile {
+		ninetyeighthpercentile = d.PercentileNearestRank(98)
+		metric = createMetric(ninetyeighthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("98%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyFifthPercentile {
+		ninetyfifthpercentile = d.PercentileNearestRank(95)
+		metric = createMetric(ninetyfifthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("95%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyNinthPercentile {
+		ninetyninthpercentile = d.PercentileNearestRank(99)
+		metric = createMetric(ninetyninthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("99%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.Skewness {
+		skewness = d.Skewness(mean, standarddeviation)
+		metric = createMetric(skewness, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("skewness")
+		results = append(results, metric)
+	}
+
+	if flag.Kurtosis {
+		kurtosis = d.Kurtosis(mean, standarddeviation)
+		metric = createMetric(kurtosis, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("kurtosis")
+		results = append(results, metric)
+	}
+
+	if flag.Trimean {
+		trimean = d.Trimean(firstquartile, median, thirdquartile)
+		metric = createMetric(trimean, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("trimean")
+		results = append(results, metric)
+	}
+	return results, err
+}
+
```

Comment:
The file start to get big starting from this point. Maybe we can add all the dataBuffer methods in another file.